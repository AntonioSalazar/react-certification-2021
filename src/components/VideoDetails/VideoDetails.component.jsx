import React, { useContext } from 'react';
import { useParams } from 'react-router';
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import './VideoDetails.styles.css';
const VideoDetails = () => {
  const { id } = useParams();
  const { videos } = useContext(VideoListContext);
  const similarVideos = videos.slice(0, 8);
  console.log(similarVideos);
  return (
    <div className="video-details-container">
      <div className="video-details-selected-video">
        <div className="video-responsive">
          <iframe
            src={`https://www.youtube.com/embed/${id}`}
            frameBorder="0"
            allowFullScreen
            title="test"
          ></iframe>
        </div>
      </div>
      <div className="video-details-similar-videos">
        <h1>placeholder content</h1>
      </div>
    </div>
  );
};

export default VideoDetails;
