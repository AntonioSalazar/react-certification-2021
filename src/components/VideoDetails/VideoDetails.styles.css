.video-details-container {
  padding: 2rem;
  display: flex;
}

.video-details-selected-video {
  width: 70%;
  height: 85vh;
}

iframe {
  width: 100%;
  height: 80%;
}

.video-details-selected-video__text {
  display: flex;
  flex-direction: column;
}

.video-details-similar-videos {
  width: 30%;
}

.title_like_btns {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 2rem;
}

.reaction-btns {
  width: 450px;
  display: flex;
  justify-content: space-around;
}

.reaction-btns i {
  margin: 0 2rem;
  font-size: 2rem;
}

.reaction-btns button {
  border: 1px solid #fff;
  padding: 1rem;
  border-radius: 4px;
  cursor: pointer;
  transition: all 0.3s ease;
}

.reaction-btns button:hover {
  box-shadow: 10px 10px 24px 0px rgba(0, 0, 0, 0.75);
}

@media only screen and (max-width: 1100px) {
  .reaction-btns {
    display: flex;
    flex-direction: column;
    width: 350px;
    justify-content: space-between;
    align-items: center;
  }

  .reaction-btns button {
    padding: 0.5rem;
    margin: 1rem 0;
  }
}

@media only screen and (max-width: 700px) {
  .video-details-container {
    display: flex;
    flex-direction: column;
  }
  .video-details-selected-video {
    margin-bottom: 3rem;
    width: 100%;
  }

  .video-details-similar-videos {
    width: 100%;
  }
}
