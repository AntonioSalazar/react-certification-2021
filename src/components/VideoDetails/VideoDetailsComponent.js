import React, { useContext } from 'react';
import { useParams } from 'react-router';
import Swal from 'sweetalert2';
import { useAuth } from '../../providers/Auth/Auth.provider';
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import { useFetch } from '../../utils/hooks/useFetch';
import RecommendedVideos from '../RecommendedVideos';
import { VideoDetailsContainer, DescriptionSpan, ReactionBtn, TitleSpan, VideoDetailsSelectedVideo, SelectedVideoTextContainer, ReactionBtnsContainer, ReactionBtnPosition, SimilarVideosContainer, Iframe } from './VideoDetailsComponent.styled';

const VideoDetails = () => {
  const { id } = useParams();
  const { singleVideo } = useFetch(id);
  const { authenticated } = useAuth();
  const { addToFavorites, videosState, removeFromFavorites } = useContext(VideoListContext);
  const { posts } = videosState;
  const { favoriteVideos } = videosState;

  const checkIfVideoIsFavorite = (id, video) => {
    if (favoriteVideos.some((eachVideo) => eachVideo.id === id)) {
      return;
    } else {
      addToFavorites(video);
    }
  };

  return (
    <VideoDetailsContainer >
      <VideoDetailsSelectedVideo >
        <Iframe
          src={`https://www.youtube.com/embed/${id}`}
          frameBorder="0"
          allowFullScreen
          title="test"
        ></Iframe>
        {singleVideo.length > 0 && (
          <SelectedVideoTextContainer>
            <ReactionBtnsContainer >
              <TitleSpan>{singleVideo[0].snippet.title}</TitleSpan>
              {authenticated ? (
                <ReactionBtnPosition>
                  {favoriteVideos.some((eachVid) => eachVid.id === id) ? (
                    <ReactionBtn type="button" onClick={() => removeFromFavorites(id)}>
                      <i className="fas fa-thumbs-down" />
                    </ReactionBtn>
                  ) : (
                    <ReactionBtn
                      type="button"
                      onClick={() => checkIfVideoIsFavorite(id, singleVideo[0])}
                    >
                      <i className="fas fa-thumbs-up" />
                    </ReactionBtn>
                  )}
                </ReactionBtnPosition>
              ) : (
                <ReactionBtn
                  type="button"
                  onClick={() =>
                    Swal.fire({
                      title: 'Oops...',
                      text: 'Log In if you want to add this video to your favorites',
                    })
                  }
                >
                  <i className="fas fa-thumbs-up" style={{ marginRight: '1rem' }} />
                  Like
                </ReactionBtn>
              )}
            </ReactionBtnsContainer>
            <DescriptionSpan>{singleVideo[0].snippet.description}</DescriptionSpan>
          </SelectedVideoTextContainer>
        )}
      </VideoDetailsSelectedVideo>
      <SimilarVideosContainer>
        <RecommendedVideos videos={posts} />
      </SimilarVideosContainer>
    </VideoDetailsContainer>
  );
};

export default VideoDetails;
