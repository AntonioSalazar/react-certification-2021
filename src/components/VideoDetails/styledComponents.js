import styled from 'styled-components';

export const TitleSpan = styled.span`
  font-size: 3rem;
  font-weight: bold;
`;
TitleSpan.displayName = 'TitleSpan';

export const DescriptionSpan = styled.span`
  font-size: 1.6rem;

  @media only screen and (max-width: 700px) {
    display: none;
    margin-bottom: 3rem;
  }
`;
DescriptionSpan.displayName = 'DescriptionSpan';