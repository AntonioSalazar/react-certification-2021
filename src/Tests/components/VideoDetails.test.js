import React from 'react';
import { mount } from 'enzyme/build';
import VideoDetails from '../../components/VideoDetails/VideoDetails.component';
import { MemoryRouter, Route } from 'react-router';
import { videoList } from '../mockData/videosList'

describe('Tests of the VideoDetials component', () => {
//     test('Should render the VideoDetails component correctly', () => {
//     const wrapper = mount( 
//         <MemoryRouter
//             initialEntries={['/video/Po3VwR_NNGk']}
//         >
//             <Route path="/video/:id" >
//                 <VideoDetails videoList />
//             </Route> 
//         </MemoryRouter>)

//     expect(wrapper).toMatchSnapshot()
// })
    
})
