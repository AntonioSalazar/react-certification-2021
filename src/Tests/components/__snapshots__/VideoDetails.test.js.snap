// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of the VideoDetials component Should render the VideoDetails component correctly 1`] = `
<div
  className="video-details-container"
>
  <div
    className="video-details-selected-video"
  >
    <iframe
      allowFullScreen={true}
      frameBorder="0"
      src="https://www.youtube.com/embed/Po3VwR_NNGk"
      title="test"
    />
  </div>
</div>
`;
